Cannot set `path` when using fields in CRUD

Consider code like this (simplified):
```html
<vaadin-crud>
  <vaadin-form-layout>
    <vaadin-text-field label="First name" path="firstName" required></vaadin-text-field>
  </vaadin-form-layout>
</vaadin-crud>
```

When translating to React, there's no `path` on `TextField`.

Version `"@hilla/react-components": "2.0.2"`.

The issue is caused by missing type definitions for the `path` property. As a temporary workaround, ignoring the error from TypeScript could help:

```tsx
<Crud items={crudData}>
  {/* @ts-expect-error TS2322: missing TypeScript definition for the "path" property */}
  <TextField path="name"></TextField>
</Crud>
````

--- a/.github/workflows/validation.yml
+++ b/.github/workflows/validation.yml
@@ -2,12 +2,12 @@ name: Validation
 
 on:
   push:
-    branches: main
+    branches: ['main', '1.3', '2.0']
     paths-ignore:
       - 'README.md'
       - '.github/ISSUE_TEMPLATE/*'
   pull_request:
-    branches: main
+    branches: ['main', '1.3', '2.0']
     paths-ignore:
       - 'README.md'
       - '.github/ISSUE_TEMPLATE/*'

--- a/scripts/cherryPick.cjs
+++ b/scripts/cherryPick.cjs
@@ -13,7 +13,7 @@
  */
 
 const axios = require('axios');
-const https = require("https");
+const https = require('https');
 const exec = require('util').promisify(require('child_process').exec);
 const { exe } = require('child_process');
 
@@ -25,31 +25,30 @@ let arrBranch = [];
 let arrUser = [];
 let arrMergedBy = [];
 
-const repo = "vaadin/react-components";
+const repo = 'vaadin/react-components';
 const token = process.env['GITHUB_TOKEN'];
 if (!token) {
   console.log(`GITHUB_TOKEN is not set, skipping PR creation`);
   process.exit(1);
 }
 
-async function getAllCommits(){
+async function getAllCommits() {
   let url = `https://api.github.com/repos/${repo}/pulls?state=closed&sort=updated&direction=desc&per_page=100`;
   try {
     const options = {
-      headers:
-      {
+      headers: {
         'User-Agent': 'Vaadin Cherry Pick',
-        'Authorization': `token ${token}`,
+        Authorization: `token ${token}`,
         'Content-Type': 'application/json',
-      }
+      },
     };
 
     res = await axios.get(url, options);
     data = res.data;
-    data = data.filter(da => da.labels.length > 0 && da.merged_at !== null);
+    data = data.filter((da) => da.labels.length > 0 && da.merged_at !== null);
 
     if (data.length === 0) {
-      console.log("No commits needs to be picked.");
+      console.log('No commits needs to be picked.');
       process.exit(0);
     }
     return data;
@@ -59,15 +58,14 @@ async function getAllCommits(){
   }
 }
 
-async function getCommit(commitURL){
+async function getCommit(commitURL) {
   try {
     const options = {
-      headers:
-      {
+      headers: {
         'User-Agent': 'Vaadin Cherry Pick',
-        'Authorization': `token ${token}`,
+        Authorization: `token ${token}`,
         'Content-Type': 'application/json',
-      }
+      },
     };
 
     res = await axios.get(commitURL, options);
@@ -80,24 +78,31 @@ async function getCommit(commitURL){
   }
 }
 
-async function filterCommits(commits){
+async function filterCommits(commits) {
   for (let commit of commits) {
     let target = false;
     let picked = false;
-    for (let label of commit.labels){
-      if(label.name.includes("target/")){
+    for (let label of commit.labels) {
+      if (label.name.includes('target/')) {
         target = true;
       }
-      if(label.name.includes("cherry-picked") || label.name.includes("need to pick manually")){
+      if (label.name.includes('cherry-picked') || label.name.includes('need to pick manually')) {
         picked = true;
       }
     }
-    if(target === true && picked === false){
+    if (target === true && picked === false) {
       let singleCommit = await getCommit(commit.url);
-      commit.labels.forEach(label => {
+      commit.labels.forEach((label) => {
         let branch = /target\/(.*)/.exec(label.name);
-        if (branch){
-          console.log(commit.number, commit.user.login, commit.url, commit.merge_commit_sha, branch[1], singleCommit.merged_by.login);
+        if (branch) {
+          console.log(
+            commit.number,
+            commit.user.login,
+            commit.url,
+            commit.merge_commit_sha,
+            branch[1],
+            singleCommit.merged_by.login,
+          );
           arrPR.push(commit.number);
           arrSHA.push(commit.merge_commit_sha);
           arrURL.push(commit.url);
@@ -106,13 +111,13 @@ async function filterCommits(commits){
           arrUser.push(`@${commit.user.login}`);
           arrMergedBy.push(`@${singleCommit.merged_by.login}`);
         }
-      })
+      });
     }
   }
 }
 
-async function cherryPickCommits(){
-  for(let i=arrPR.length-1; i>=0; i--){
+async function cherryPickCommits() {
+  for (let i = arrPR.length - 1; i >= 0; i--) {
     let branchName = `cherry-pick-${arrPR[i]}-to-${arrBranch[i]}-${Date.now()}`;
 
     await exec('git checkout main');
@@ -120,15 +125,15 @@ async function cherryPickCommits(){
     await exec(`git checkout ${arrBranch[i]}`);
     await exec(`git reset --hard origin/${arrBranch[i]}`);
 
-    try{
+    try {
       await exec(`git checkout -b ${branchName}`);
     } catch (err) {
       console.error(`Cannot Create Branch, error : ${err}`);
       process.exit(1);
     }
 
-    try{
-      let {stdout, stderr} = await exec(`git cherry-pick ${arrSHA[i]}`);
+    try {
+      let { stdout, stderr } = await exec(`git cherry-pick ${arrSHA[i]}`);
     } catch (err) {
       console.error(`Cannot Pick the Commit:${arrSHA[i]} to ${arrBranch[i]}, error :${err}`);
       await labelCommit(arrURL[i], `need to pick manually ${arrBranch[i]}`);
@@ -147,68 +152,75 @@ async function cherryPickCommits(){
   }
 }
 
-async function labelCommit(url, label){
-  let issueURL = url.replace("pulls", "issues") + "/labels";
+async function labelCommit(url, label) {
+  let issueURL = url.replace('pulls', 'issues') + '/labels';
   const options = {
-    headers:{
+    headers: {
       'User-Agent': 'Vaadin Cherry Pick',
-      'Authorization': `token ${token}`,
-    }
+      Authorization: `token ${token}`,
+    },
   };
 
-  await axios.post(issueURL, {"labels":[label]}, options);
+  await axios.post(issueURL, { labels: [label] }, options);
 }
 
-async function postComment(url, userName, mergedBy, branch, message){
-  let issueURL = url.replace("pulls", "issues") + "/comments";
+async function postComment(url, userName, mergedBy, branch, message) {
+  let issueURL = url.replace('pulls', 'issues') + '/comments';
   const options = {
-    headers:{
+    headers: {
       'User-Agent': 'Vaadin Cherry Pick',
-      'Authorization': `token ${token}`,
-    }
+      Authorization: `token ${token}`,
+    },
   };
 
-  await axios.post(issueURL, {"body":`Hi ${userName} and ${mergedBy}, when i performed cherry-pick to this commit to ${branch}, i have encountered the following issue. Can you take a look and pick it manually?\n Error Message:\n ${message}`}, options);
+  await axios.post(
+    issueURL,
+    {
+      body: `Hi ${userName} and ${mergedBy}, when i performed cherry-pick to this commit to ${branch}, i have encountered the following issue. Can you take a look and pick it manually?\n Error Message:\n ${message}`,
+    },
+    options,
+  );
 }
 
+async function createPR(title, head, base) {
+  const payload = { title, head, base };
 
-async function createPR(title, head, base){
-  const payload = {title, head, base};
-
-  return new Promise(resolve => {
-    const content = JSON.stringify({ title, head, base }, null, 1)
-    const req = https.request({
-      method: 'POST',
-      hostname: 'api.github.com',
-      path: `/repos/${repo}/pulls`,
-      headers: {
-        'Authorization': `token ${token}`,
-        'User-Agent': 'Vaadin Cherry Pick',
-        'Content-Type': 'application/json',
-        'Content-Length': content.length,
+  return new Promise((resolve) => {
+    const content = JSON.stringify({ title, head, base }, null, 1);
+    const req = https.request(
+      {
+        method: 'POST',
+        hostname: 'api.github.com',
+        path: `/repos/${repo}/pulls`,
+        headers: {
+          Authorization: `token ${token}`,
+          'User-Agent': 'Vaadin Cherry Pick',
+          'Content-Type': 'application/json',
+          'Content-Length': content.length,
+        },
+        body: content,
       },
-      body: content
-    }, res => {
-      let body = "";
-      res.on("data", data => {
-        body += data;
-      });
-      res.on("end", () => {
-        resolve(body);
-      });
-    });
-    req.write(content)
-  }).then(body => {
+      (res) => {
+        let body = '';
+        res.on('data', (data) => {
+          body += data;
+        });
+        res.on('end', () => {
+          resolve(body);
+        });
+      },
+    );
+    req.write(content);
+  }).then((body) => {
     const resp = JSON.parse(body);
     console.log(`Created PR '${title}' ${resp.url}`);
   });
 }
 
-async function main(){
+async function main() {
   let allCommits = await getAllCommits();
   await filterCommits(allCommits);
   await cherryPickCommits();
-
 }
 
 main();

--- a/scripts/utils/misc.ts
+++ b/scripts/utils/misc.ts
@@ -35,9 +35,9 @@ export function stripPrefix(str: string) {
  *
  * @returns The camel case class name without Vaadin prefix.
  */
-export function convertElementNameToClassName(elementName: string) {
+export function convertElementNameToClassName(elementName: string): string {
   if (elementToClassNamingConventionViolations.has(elementName)) {
-    return elementToClassNamingConventionViolations.get(elementName);
+    return elementToClassNamingConventionViolations.get(elementName)!;
   }
 
   const conventionalClassName = stripPrefix(camelCase(elementName));

--- a/src/Crud.ts
+++ b/src/Crud.ts
@@ -1,33 +1,24 @@
 export * from './generated/Crud.js';
 
 /**
- * HTML attributes that the `<Crud />` component recognizes on the fields
- * of a custom form.
- */
-export type CrudFieldProps = Readonly<{
-  /**
-   * A key or path for the value of the particular field in the CRUD item
-   * objects.
-   */
-  path: string;
-}>;
-
-/**
- * A helper function that allows declaring HTML attributes on the fields of
- * a custom form in the `<Crud />` component.
- *
+ * A helper function that allows declaring the value path (or key) on
+ * the fields in a custom form of a `<Crud />` component.
+ * 
  * ### Usage
- *
+ * 
  * ```tsx
  * <Crud>
  *   <FormLayout slot="form">
- *     <TextField label="Name" {...{crudFieldProps({path: 'name'})}}></TextField>
+ *     <TextField label="Name" {...crudFieldPath('name')} />
  *   </FormLayout>
  * </Crud>
  * ```
- *
- * @param props HTML attributes of the field
+ * 
+ * @param path The path of the particular field's value in the CRUD
+ * item objects.
+ * 
+ * @returns object with HTML attribute values recognized on a CRUD form field.
  */
-export function crudFieldProps(props: CrudFieldProps) {
-  return props as {};
+export function crudFieldPath(path: string): Record<string, string> {
+  return {path} as Record<string, string>;
 }

--- a/src/Crud.ts
+++ b/src/Crud.ts
@@ -1,1 +1,33 @@
 export * from './generated/Crud.js';
+
+/**
+ * HTML attributes that the `<Crud />` component recognizes on the fields
+ * of a custom form.
+ */
+export type CrudFieldProps = Readonly<{
+  /**
+   * A key or path for the value of the particular field in the CRUD item
+   * objects.
+   */
+  path: string;
+}>;
+
+/**
+ * A helper function that allows declaring HTML attributes on the fields of
+ * a custom form in the `<Crud />` component.
+ *
+ * ### Usage
+ *
+ * ```tsx
+ * <Crud>
+ *   <FormLayout slot="form">
+ *     <TextField label="Name" {...{crudFieldProps({path: 'name'})}}></TextField>
+ *   </FormLayout>
+ * </Crud>
+ * ```
+ *
+ * @param props HTML attributes of the field
+ */
+export function crudFieldProps(props: CrudFieldProps) {
+  return props as {};
+}

--- a/src/Crud.ts
+++ b/src/Crud.ts
@@ -9,7 +9,7 @@ export * from './generated/Crud.js';
  * ```tsx
  * <Crud>
  *   <FormLayout slot="form">
- *     <TextField label="Name" {...crudFieldPath('name')} />
+ *     <TextField label="Name" {...crudPath('name')} />
  *   </FormLayout>
  * </Crud>
  * ```
@@ -19,6 +19,6 @@ export * from './generated/Crud.js';
  * 
  * @returns object with HTML attribute values recognized on a CRUD form field.
  */
-export function crudFieldPath(path: string): Record<string, string> {
+export function crudPath(path: string): Record<string, string> {
   return {path} as Record<string, string>;
 }

--- a/src/Crud.ts
+++ b/src/Crud.ts
@@ -3,9 +3,9 @@ export * from './generated/Crud.js';
 /**
  * A helper function that allows declaring the value path (or key) on
  * the fields in a custom form of a `<Crud />` component.
- * 
+ *
  * ### Usage
- * 
+ *
  * ```tsx
  * <Crud>
  *   <FormLayout slot="form">
@@ -13,12 +13,12 @@ export * from './generated/Crud.js';
  *   </FormLayout>
  * </Crud>
  * ```
- * 
+ *
  * @param path The path of the particular field's value in the CRUD
  * item objects.
- * 
+ *
  * @returns object with HTML attribute values recognized on a CRUD form field.
  */
 export function crudPath(path: string): Record<string, string> {
-  return {path} as Record<string, string>;
+  return { path } as Record<string, string>;
 }

--- a/test/MultiSelectComboBox.spec.tsx
+++ b/test/MultiSelectComboBox.spec.tsx
@@ -9,7 +9,10 @@ useChaiPlugin(chaiDom);
 describe('MultiSelectComboBox', () => {
   const overlayTag = 'vaadin-multi-select-combo-box-overlay';
 
-  const [ref, catcher] = createOverlayCloseCatcher<MultiSelectComboBoxElement>(overlayTag, (ref) => ref.close());
+  const [ref, catcher] = createOverlayCloseCatcher<MultiSelectComboBoxElement>(
+    overlayTag,
+    (ref) => (ref.opened = false),
+  );
 
   afterEach(cleanup);
   afterEach(catcher);

--- a/test/kitchen-sink/Row3.tsx
+++ b/test/kitchen-sink/Row3.tsx
@@ -1,10 +1,14 @@
 import { BoardRow } from '../../src/BoardRow.js';
+import { Checkbox } from '../../src/Checkbox.js';
+import { ComboBox } from '../../src/ComboBox.js';
 import { ContextMenu } from '../../src/ContextMenu.js';
 import { CookieConsent } from '../../src/CookieConsent.js';
-import { Crud } from '../../src/Crud.js';
+import { Crud, crudFieldProps } from '../../src/Crud.js';
+import { FormLayout } from '../../src/FormLayout.js';
 import { Item } from '../../src/Item.js';
 import { ListBox } from '../../src/ListBox.js';
-import { crudData } from './data.js';
+import { TextField } from '../../src/TextField.js';
+import { CrudRole, crudData } from './data.js';
 
 const menuItems = [{ text: 'Edit' }, { text: 'Delete' }];
 
@@ -17,7 +21,13 @@ export default function Row3() {
         </ListBox>
       </ContextMenu>
       <CookieConsent position="bottom-right"></CookieConsent>
-      <Crud items={crudData}></Crud>
+      <Crud items={crudData}>
+        <FormLayout slot="form">
+          <TextField label="Name" {...crudFieldProps({ path: 'name' })} />
+          <ComboBox label="Role" items={Object.values(CrudRole)} {...crudFieldProps({ path: 'role' })} />
+          <Checkbox label="Active" {...crudFieldProps({ path: 'active' })} />
+        </FormLayout>
+      </Crud>
     </BoardRow>
   );
 }

--- a/test/kitchen-sink/Row3.tsx
+++ b/test/kitchen-sink/Row3.tsx
@@ -3,7 +3,7 @@ import { Checkbox } from '../../src/Checkbox.js';
 import { ComboBox } from '../../src/ComboBox.js';
 import { ContextMenu } from '../../src/ContextMenu.js';
 import { CookieConsent } from '../../src/CookieConsent.js';
-import { Crud, crudFieldProps } from '../../src/Crud.js';
+import { Crud, crudFieldPath } from '../../src/Crud.js';
 import { FormLayout } from '../../src/FormLayout.js';
 import { Item } from '../../src/Item.js';
 import { ListBox } from '../../src/ListBox.js';
@@ -23,9 +23,9 @@ export default function Row3() {
       <CookieConsent position="bottom-right"></CookieConsent>
       <Crud items={crudData}>
         <FormLayout slot="form">
-          <TextField label="Name" {...crudFieldProps({ path: 'name' })} />
-          <ComboBox label="Role" items={Object.values(CrudRole)} {...crudFieldProps({ path: 'role' })} />
-          <Checkbox label="Active" {...crudFieldProps({ path: 'active' })} />
+          <TextField label="Name" {...crudFieldPath('name')} />
+          <ComboBox label="Role" items={Object.values(CrudRole)} {...crudFieldPath('role')} />
+          <Checkbox label="Active" {...crudFieldPath('active')} />
         </FormLayout>
       </Crud>
     </BoardRow>

--- a/test/kitchen-sink/Row3.tsx
+++ b/test/kitchen-sink/Row3.tsx
@@ -3,7 +3,7 @@ import { Checkbox } from '../../src/Checkbox.js';
 import { ComboBox } from '../../src/ComboBox.js';
 import { ContextMenu } from '../../src/ContextMenu.js';
 import { CookieConsent } from '../../src/CookieConsent.js';
-import { Crud, crudFieldPath } from '../../src/Crud.js';
+import { Crud, crudPath } from '../../src/Crud.js';
 import { FormLayout } from '../../src/FormLayout.js';
 import { Item } from '../../src/Item.js';
 import { ListBox } from '../../src/ListBox.js';
@@ -23,9 +23,9 @@ export default function Row3() {
       <CookieConsent position="bottom-right"></CookieConsent>
       <Crud items={crudData}>
         <FormLayout slot="form">
-          <TextField label="Name" {...crudFieldPath('name')} />
-          <ComboBox label="Role" items={Object.values(CrudRole)} {...crudFieldPath('role')} />
-          <Checkbox label="Active" {...crudFieldPath('active')} />
+          <TextField label="Name" {...crudPath('name')} />
+          <ComboBox label="Role" items={Object.values(CrudRole)} {...crudPath('role')} />
+          <Checkbox label="Active" {...crudPath('active')} />
         </FormLayout>
       </Crud>
     </BoardRow>

--- a/test/kitchen-sink/data.ts
+++ b/test/kitchen-sink/data.ts
@@ -1,4 +1,4 @@
-enum CrudRole {
+export enum CrudRole {
   ADMIN = 'admin',
   USER = 'user',
 }

